### ISSUE TITLE
How should I reproduce the process of generating black images using SD

### ISSUE THREAD

**Issue by TheHonestBob** [huggingface/diffusers#5803]
### Describe the bug

When I use SD for prediction in a production environment, I occasionally encounter black images. In order to observe which model is causing the problem, I try to record the input image, prompt, specify the generator, and record the corresponding seed, but it is not possible to reproduce the black image. Although the reproduced image is fixed, is there any way to reproduce it so that I can track the cause of the problem

### Reproduction
```python
lora_path = r'lora_weight/ma 3.0.safetensors'
image_device = 'cuda:1'
control_net = ControlNetModel.from_pretrained("control_weight/sd-controlnet-canny",
                                              torch_dtype=torch.float16,
                                              use_safetensors=True)
vae = AutoencoderKL.from_single_file("vae_weight/animevae.pt",
                                     config_file=r'config/v1-inference.yaml',
                                     torch_dtype=torch.float32,
                                     local_files_only=True,
                                     use_safetensors=True)
pipe = StableDiffusionControlNetPipeline.from_single_file(
    r'sd_weight/AWPainting_v1.0.safetensors',
    controlnet=control_net,
    vae=vae,
    torch_dtype=torch.float16,
    cache_dir='sd_weight',
    local_files_only=True,
    original_config_file=r'config/v1-inference.yaml',
    load_safety_checker=None,
    use_safetensors=True,
    scheduler_type='euler'
).to(image_device)

pipe.load_lora_weights(lora_path)
prompt = 'woman red dress long black hair blurry close-up no clothes sensual mysterious low-key lighting artistic provocative cinematic high resolution digital art. power_lines,halftone_background,cityscape,rolling_eyes,robe_of_blending'
negative_prompt = '''
low quality, bad anatomy, mistakes, missing fingers, extra fingers, fewer fingers, extra limbs, multiple breasts,
 deformities, skin spots, acne, age spots, skin blemishes, bad face, deformed fingers, EasyNegative,No nudity, No porn.
 ,Don't dress revealingly
'''

img = r'black_image_-72132.jpg'
user_img = np.asarray(load_image(img))
print(user_img.shape)
image = cv2.Canny(user_img, 100, 200)
image = image[:, :, None]
image = np.concatenate([image, image, image], axis=2)
image = Image.fromarray(image)

sd_parameter = {
        'image': image,
        'prompt': prompt,
        'negative_prompt': negative_prompt,
        'height': 960,
        'width': 544,
        'num_inference_steps': 20,
        # 每个prompt固定生成一张
        'num_images_per_prompt': 1,
        'guidance_scale': 8.0,
        'controlnet_conditioning_scale': 0.6,
        'control_guidance_start': 0.0,
        'control_guidance_end': 1.0,
        'generator': torch.Generator(device=image_device).manual_seed(-72132)
    }

images = pipe(**sd_parameter).images
images[0].save(r'error.jpg')
```
### Logs

```shell
python3.10/site-packages/diffusers/image_processor.py:88: RuntimeWarning: invalid value encountered in cast images = (images * 255).round().astype("uint8")
```

### System Info

ubuntu 20
GPU 3090
diffusers 0.23.0.dev0

**Comment by nickkolok** [comment]
Looks like an NSFW checker issue.

Firstly, you should not use "no" in the negative prompt. Use `porn` in the negative instead of `No porn`.

Secondly, have you tried to generate images of something presunable SFW, like a cat or a bicycle?

Finally, you may want to change vae to another - they say sometimes it helps.

**Comment by yiyixuxu** [comment]
hi @TheHonestBob 

I don't think I completely understand here - you were getting black image with the script you provided, and when you run the same script again, you got a different output?

**Comment by TheHonestBob** [comment]
> Looks like an NSFW checker issue.
> 
> Firstly, you should not use "no" in the negative prompt. Use `porn` in the negative instead of `No porn`.
> 
> Secondly, have you tried to generate images of something presunable SFW, like a cat or a bicycle?
> 
> Finally, you may want to change vae to another - they say sometimes it helps.

thanks for your replay，Firstly,I set load_safety_checker=None, So I don't think it's a problem with NSFW. Secondly, Thank you for pointing out the error in my code， Finally I will try a lot of tests on different vaes to see if there will be black images again

**Comment by TheHonestBob** [comment]
> hi @TheHonestBob
> 
> I don't think I completely understand here - you were getting black image with the script you provided, and when you run the same script again, you got a different output?

yes, when use my script in flask server, I get black image, when I reproduce it, I coundn't get black image , so that I can't debug code where is  error，even though I record prompt、input image、seed, same system、GPU，when I reproduce, I get same result but not black

**Comment by yiyixuxu** [comment]
it this also a multithreaded environment?

**Comment by TheHonestBob** [comment]
> it this also a multithreaded environment?

It's not multithreaded. Later, I replaced the vae several times, and after testing the final generation of 3840 images, I found that there was a problem with the vae I initially selected in my scenario. Although I don't know what caused the vae to have a nan value so far, my situation is indeed a problem with the vae. Is there an error in training the vae
